src/averageAges.js

```diff
@@ -37,7 +52,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const toFilterWomen = (person) => {
+    const isWoman = person.sex === 'f';
+
+    const isWithChildren = people.some(
+      human => person.name === human.mother
+    );
+
+    return withChildren ? isWoman && isWithChildren : isWoman;
```

Comment:
1 - avoid writing ternary operators in one line, it's harder to read
2 - `isWoman` duplicated, move it out of ternary